src/main/java/org/codeforamerica/messaging/services/SmsService.java

```diff
@@ -16,8 +27,22 @@ public SmsService(TwilioGateway twilioGateway, SmsMessageRepository smsMessageRe
         this.smsMessageRepository = smsMessageRepository;
     }
 
-    public SmsMessage sendSmsMessage(String to, String body) throws MessageSendException {
-        SmsMessage smsMessage = twilioGateway.sendMessage(to, body);
+    public SmsMessage sendSmsMessage(String toPhone, String body) throws MessageSendException {
+        if (duplicate(toPhone, body)) {
+            log.error("Skipping duplicate SMS");
+            throw new DuplicateMessageException();
+        }
+        SmsMessage smsMessage = twilioGateway.sendMessage(toPhone, body);
         return smsMessageRepository.save(smsMessage);
     }
+
+    private boolean duplicate(String toPhone, String body) {
+        List<SmsMessage> matchingSmsMessagesInWindow = smsMessageRepository.findAllByToPhoneAndBodyAndUpdateTimestampAfter(
+            toPhone, body, OffsetDateTime.now().with(DUPLICATE_MESSAGE_SUPPRESSION_WINDOW));
```

Comment:
IMO, `OffsetDateTime.now().minusHours(DUPLICATE_MESSAGE_SUPPRESSION_WINDOW)` where DUPLICATE_MESSAGE_SUPPRESSION_WINDOW=23 would be easier to understand.

Reply:
The downside with that is the unit of "hours" is not apparent in the variable, unless named DUPLICATE_MESSAGE_SUPPRESSION_HOURS, and then also what if the unit is changed later. I get it though that the lambda is ugly and hard to read, so I'll change it since the other concerns are unlikely.